dfaker.py

```diff
@@ -0,0 +1,387 @@
+from datetime import datetime, timedelta, tzinfo
+import time
+from pytz import timezone
+import pytz
+import uuid
+import json
+import argparse 
+import sys
+import cbg_equation
+import statsmodels.api as sm
+
+import random
+import numpy as np 
+
+params = {
+	'year' : 2015, #default datetime settings
+	'month' : 1,
+	'day' : 1,
+	'hour' : 0,
+	'minute' : 0,
+	'zone' : 'US/Pacific', #default zone
+	'num_days' : 30, #default days to generate data for
+	'file' : 'device-data.json', #default json file name 
+	'minify' : False, #compact storage option false by default 
+	'gaps' : False, #randomized gaps in data, off by default 
+	'smbg_freq' : 6 #default fingersticks per day
+}
+
+def parse(args, params):
+	if args.date:
+		try:
+			params['year'] = int(args.date[0:4])
+			params['month'] = int(args.date[5:7]) 
+			params['day'] = int(args.date[8:])
+		
+			if (params['month'] not in range(1, 13) 
+					or params['day'] not in range(1, 32) 
+					or params['year'] not in range(1900, 2100) 
+					or len(args.date) != 10):
+				print('Invalid range: {:s}. Please enter date as YYYY-MM-DD'.format(args.date))
+				sys.exit(1)
+		except:		
+			print('Wrong date format: {:s}. Please enter date as YYYY-MM-DD'.format(args.date))
+			sys.exit(1)
+
+	if args.time:
```

Comment:
Same feedback as above with trying `datetime.strptime`. (BTW, if you're wondering, 'strptime' stands for string (`str`) parse (`p`) time. And the opposite one is `strftime` = string (`str`) format (`f`) time. For a long time those were just magic confusing letter combinations to me, but maybe I'm just slow...)